### ISSUE TITLE
Duplicated logs in Loki with Loki Canary and Promtail in K8s

### ISSUE THREAD

**Issue by dragoangel** [grafana/loki#12059]
**Describe the bug**
Same logs appears twice only for loki namespace itself. Once they are using proper labels, parsing etc, and second one - not.

I see labels in wrong case that never was been set via config of Promtail like `app_kubernetes_io_part_of`, but they exists. Don't understand how this possible, I see that Promtail doesn't honnor `droplabel` and all other settings for them, for example - if my pod has `promtail.io/exclude` it would be dropped from Loki, except cases with mentioned duplicated broken records, they will not.

**To Reproduce**
1. Setup Loki with `test.enabled` and Promtail via Helm, both with 2.9.3
2. Check logs in Grafana for Loki namespace
3. Verify that multiply pods has deployed on same node - more pods in Loki namespace will be on same node - much more impact from this bug will happen

**Expected behavior**
Logs will be collected only once by Promtail, and will be honor all settings and stages. F.e. `docker` stage parse docker's `json` logs and outputs them as strings with proper timestamp etc.

**Environment:**
 - Infrastructure: Kubernetes
 - Deployment tool: helm

**Screenshots, Promtail config, or terminal output**

![image](https://github.com/grafana/loki/assets/1865999/d3a0f226-d906-47d2-a535-5949aba4e7c4)

Promtail has next settings:
```yaml
global:
  file_watch_config:
    min_poll_frequency: 250ms
    max_poll_frequency: 250ms
server:
  http_listen_network: tcp
  http_listen_address: ""
  http_listen_port: 3101
  http_listen_conn_limit: 0
  grpc_listen_network: tcp
  grpc_listen_address: ""
  grpc_listen_port: 9095
  grpc_listen_conn_limit: 0
  tls_cipher_suites: ""
  tls_min_version: ""
  http_tls_config:
    cert_file: ""
    key_file: ""
    client_auth_type: ""
    client_ca_file: ""
  grpc_tls_config:
    cert_file: ""
    key_file: ""
    client_auth_type: ""
    client_ca_file: ""
  register_instrumentation: true
  graceful_shutdown_timeout: 30s
  http_server_read_timeout: 30s
  http_server_write_timeout: 30s
  http_server_idle_timeout: 2m0s
  grpc_server_max_recv_msg_size: 4194304
  grpc_server_max_send_msg_size: 4194304
  grpc_server_max_concurrent_streams: 100
  grpc_server_max_connection_idle: 2562047h47m16.854775807s
  grpc_server_max_connection_age: 2562047h47m16.854775807s
  grpc_server_max_connection_age_grace: 2562047h47m16.854775807s
  grpc_server_keepalive_time: 2h0m0s
  grpc_server_keepalive_timeout: 20s
  grpc_server_min_time_between_pings: 5m0s
  grpc_server_ping_without_stream_allowed: false
  log_format: logfmt
  log_level: info
  log_source_ips_enabled: false
  log_source_ips_header: ""
  log_source_ips_regex: ""
  log_request_headers: false
  log_request_at_info_level_enabled: false
  log_request_exclude_headers_list: ""
  http_path_prefix: ""
  external_url: ""
  health_check_target: null
  disable: false
  profiling_enabled: false
  enable_runtime_reload: false
client:
  url: ""
  batchwait: 1s
  batchsize: 1048576
  follow_redirects: false
  enable_http2: false
  backoff_config:
    min_period: 500ms
    max_period: 5m0s
    max_retries: 10
  external_labels: |
    instance: rke-node-name
  timeout: 10s
  tenant_id: ""
  drop_rate_limited_batches: false
  stream_lag_labels: ""
clients:
- url: http://loki-gateway/loki/api/v1/push
  batchwait: 1s
  batchsize: 1048576
  follow_redirects: true
  enable_http2: true
  backoff_config:
    min_period: 500ms
    max_period: 5m0s
    max_retries: 10
  external_labels: |
    instance: rke-node-name
  timeout: 10s
  tenant_id: ""
  drop_rate_limited_batches: false
  stream_lag_labels: ""
positions:
  sync_period: 10s
  filename: /run/promtail/positions.yaml
  ignore_invalid_yaml: false
scrape_configs:
- job_name: kubernetes-pods
  pipeline_stages:
  - docker: null
  - decolorize: null
  - limit:
      burst: 5000
      rate: 10000
  - drop:
      drop_counter_reason: line_too_old
      older_than: 24h
  - drop:
      drop_counter_reason: line_too_long
      longer_than: 1024kb
  - labeldrop:
    - filename
  - static_labels:
      cluster: cluster
      environment: environment
      tenant: tenant
  relabel_configs:
  - source_labels: [__meta_kubernetes_pod_controller_name]
    separator: ;
    regex: ([0-9a-z-.]+?)(-[0-9a-f]{8,10})?
    target_label: __tmp_controller_name
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_label_app_kubernetes_io_name, __meta_kubernetes_pod_label_app,
      __tmp_controller_name, __meta_kubernetes_pod_name]
    separator: ;
    regex: ^;*([^;]+)(;.*)?$
    target_label: app
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_label_app_kubernetes_io_instance, __meta_kubernetes_pod_label_instance]
    separator: ;
    regex: ^;*([^;]+)(;.*)?$
    target_label: instance
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_label_app_kubernetes_io_component, __meta_kubernetes_pod_label_component]
    separator: ;
    regex: ^;*([^;]+)(;.*)?$
    target_label: component
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_annotation_promtail_io_exclude]
    separator: ;
    regex: "true"
    replacement: $1
    action: drop
  - source_labels: [__meta_kubernetes_pod_node_name]
    separator: ;
    regex: (.*)
    target_label: instance
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_namespace]
    separator: ;
    regex: (.*)
    target_label: namespace
    replacement: $1
    action: replace
  - source_labels: [namespace, app]
    separator: /
    regex: (.*)
    target_label: job
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_name]
    separator: ;
    regex: (.*)
    target_label: pod
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_container_name]
    separator: ;
    regex: (.*)
    target_label: container
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_uid, __meta_kubernetes_pod_container_name]
    separator: /
    regex: (.*)
    target_label: __path__
    replacement: /var/log/pods/*$1/*.log
    action: replace
  - source_labels: [__meta_kubernetes_pod_annotationpresent_kubernetes_io_config_hash,
      __meta_kubernetes_pod_annotation_kubernetes_io_config_hash, __meta_kubernetes_pod_container_name]
    separator: /
    regex: true/(.*)
    target_label: __path__
    replacement: /var/log/pods/*$1/*.log
    action: replace
  - source_labels: [__meta_kubernetes_pod_container_image]
    separator: ;
    regex: (.*)
    target_label: image
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_ip]
    separator: ;
    regex: (.*)
    target_label: pod_ip
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_label_pod_template_hash]
    separator: ;
    regex: (.*)
    target_label: pod_template_hash
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_label_pod_template_generation]
    separator: ;
    regex: (.*)
    target_label: pod_template_generation
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_label_controller_revision_hash]
    separator: ;
    regex: (.*)
    target_label: controller_revision_hash
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_label_component]
    separator: ;
    regex: (.*)
    target_label: component
    replacement: $1
    action: replace
  static_configs: []
  kubernetes_sd_configs:
  - role: pod
    kubeconfig_file: ""
    follow_redirects: true
    enable_http2: true
    selectors:
    - role: pod
      field: spec.nodeName=rke-node-name
- job_name: rke
  pipeline_stages:
  - regex:
      expression: .*\/(?P<container>[a-zA-Z0-9-]+)_(?P<container_id>[a-z0-9]{64})\.log
      source: filename
  - labels:
      container: null
      container_id: null
  - docker: null
  - decolorize: null
  - limit:
      burst: 5000
      rate: 10000
  - drop:
      drop_counter_reason: line_too_old
      older_than: 24h
  - drop:
      drop_counter_reason: line_too_long
      longer_than: 1024kb
  - labeldrop:
    - filename
  - static_labels:
      cluster: cluster
      environment: environment
      tenant: tenant
  static_configs:
  - targets:
    - localhost
    labels:
      __path__: /var/lib/rancher/rke/log/*.log
      job: rancher-rke
- job_name: journal
  pipeline_stages:
  - decolorize: null
  - limit:
      burst: 5000
      rate: 10000
  - drop:
      drop_counter_reason: line_too_old
      older_than: 24h
  - drop:
      drop_counter_reason: line_too_long
      longer_than: 1024kb
  - static_labels:
      cluster: cluster
      environment: environment
      tenant: tenant
  journal:
    max_age: 12h
    json: false
    labels:
      job: systemd-journal
    path: /var/log/journal
    matches: ""
  relabel_configs:
  - source_labels: [__journal__systemd_unit]
    separator: ;
    regex: (.*)
    target_label: unit
    replacement: $1
    action: replace
  - source_labels: [__journal_priority_keyword]
    separator: ;
    regex: (.*)
    target_label: level
    replacement: $1
    action: replace
  - source_labels: [__journal_container_name]
    separator: ;
    regex: (.*)
    target_label: container
    replacement: $1
    action: replace
  - source_labels: [__journal_container_id_full]
    separator: ;
    regex: (.*)
    target_label: container_id
    replacement: $1
    action: replace
  static_configs: []
target_config:
  sync_period: 10s
  stdin: false
tracing:
  enabled: false
wal:
  enabled: false
  dir: ""
  cleanSegmentsOlderThan: 0s
  watchConfig:
    minReadFrequency: 0s
    maxReadFrequency: 0s
```

**Comment by dragoangel** [comment]
Tried enable debug logs in Promtail but doesn't get anything useful. After debuging I still can't find root case, only symptoms that increase impact:

1. disabling loki test and as result loki-canary remove unexpected logs
2. on environments with low amount of nodes, where different pods scheduled on same node, for example test env this issue has much bigger impact

**Comment by dragoangel** [comment]
Sorry, but this issue quite annoying, is there any hints what I can share to help fixing it?

**Comment by dragoangel** [comment]
I found reason why logs was duplicated. Reason is I deployed loki with Grafana Operator and self monitoring custom resource which creates loki-logs deployment that scrapes all data to loki in addition to promtail that I also deployed and configured to scrape all k8s logs. I don't need at the moment custom scraping per namespace and want just to log everything I have in k8 stdout so I disabled and removed grafana operator with it's CRDs which effectively dropped loki-logs daemonset and removed logs duplication. I assume that more details about that promtail doesn't needed if you have grafana operator would solve misunderstanding.
